index.bs

```diff
@@ -193,9 +204,18 @@ interface FileSystemHandle {
 };
 </xmp>
 
-A {{FileSystemHandle}} object represents a [=/file system entry=]. Each {{FileSystemHandle}} object is associated
-with an <dfn for=FileSystemHandle export>entry</dfn> (a [=/file system entry=]). Multiple separate objects implementing
```

Comment:
Un-defining `[=FileSystemHandle/entry=]` breaks most of this spec (and https://github.com/WICG/file-system-access), and a FileSystemHandle _does_ conceptually have an associated entry... even if getting that entry requires looking it up using path.

Is it possible to have an "associated" member (i.e. `[=FileSystemHandle/entry=]`) that's retrieved using an algorithm?

It would be something like:
```
Each {{FileSystemHandle}} object is associated with a
<dfn for=FileSystemHandle export>path</dfn> (a [=/file system path=])
which maps to an associated <dfn for=FileSystemHandle export>entry</dfn> (a [=/file system entry=])
using an [=implementation-defined=] mapping.
```

Reply:
Isn't part of the problem that it doesn't always map to an entry? I think you'd have to change all call sites anyway to take that into account and given that it seems better to have a more explicit algorithm that takes you from one to the other.

Reply:
Yeah that's what I was thinking. It seems to me like the options are:

- a path can always map to an entry, which may "exist" or not (as currently sketched out in the `getFile()` method)
- the locate method returns either a valid entry, or nothing. And then perhaps we can add a `[=FileSystemHandle/exists=]` algorithm for simplicity. The latter seems more intuitive to me. I can test this out in the next patch...

Reply:
> * the locate method returns either a valid entry, or nothing

This is sketched out in the latest patch!

Reply:
Related to my comment above on tying access checks to paths, the current chrome implementation (at least last I looked at it, which admittedly is a while ago, so maybe that has been changed already?) does not guarantee that two handles representing the same path behave the same in all aspects (since for handles acquired via DirectoryHandle.getFileHandle/getDirectoryHandle the access checks are done with the path of the original directory, rather than the path of the current handle).

I'm not saying that shouldn't be changed, but it's certainly something to keep in mind. Perhaps tying access checks to the FileSystemHandle rather than the entry would be a way around that, if we want to keep the same behavior? Although you'd have to figure out how to serialize that properly as well.

Reply:
Sorry for my confusion, but why would we need to serialize access checks if we were to move them from the "entry" to the FileSystemHandle?